ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/RecoverySuspendTest.java

```diff
@@ -0,0 +1,165 @@
+import com.hp.mwtests.ts.arjuna.resources.BytemanControlledRecord;
+import org.jboss.byteman.contrib.bmunit.BMScript;
+import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
+import org.jboss.byteman.contrib.bmunit.WithByteman;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+@WithByteman
+@BMUnitConfig(debug=true)
+public class RecoverySuspendTest {
+
+    private static RecoveryManager _manager;
+    private static RecordTypeMap _recordTypeMap;
+
+    // fields read with byteman
+    private final static int periodicRecoveryPeriod = 5;
+    private final static int recoveryBackoffPeriod = 1;
+
+    @BeforeAll
+    public static void beforeClass() {
+
+        RecoveryEnvironmentBean recoveryConfig = recoveryPropertyManager.getRecoveryEnvironmentBean();
+        String[] modules = {
+                // the module to test
+                "com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule"
+        };
+
+        // Let's go quick on this
+        recoveryConfig.setRecoveryBackoffPeriod(recoveryBackoffPeriod);
+        recoveryConfig.setPeriodicRecoveryPeriod(periodicRecoveryPeriod);
+        // don't sign off until the store is empty
+        recoveryConfig.setWaitForFinalRecovery(true);
+
+        // the test set of modules
+        recoveryConfig.setRecoveryModuleClassNames(Arrays.asList(modules));
+
+        // obtain a new RecoveryManager with the above config:
+        _manager = RecoveryManager.manager(RecoveryManager.INDIRECT_MANAGEMENT);
+        // recovery can start
+        _manager.startRecoveryManagerThread();
+
+        _recordTypeMap = new RecordTypeMap() {
+            @SuppressWarnings("unchecked")
+            public Class getRecordClass ()
+            {
+                return BytemanControlledRecord.class;
+            }
+
+            public int getType ()
+            {
+                return RecordType.USER_DEF_FIRST0;
+            }
+        };
+
+        RecordTypeManager.manager().add(_recordTypeMap);
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        // In case the Recovery Manager was suspended, this will resume it
+        _manager.resume();
+    }
+
+    @AfterAll
+    public static void afterClass() {
+        RecordTypeManager.manager().remove(_recordTypeMap);
+    }
+
+    @AfterEach
+    public void afterEach() {
+        if (!TxControl.isEnabled()) {
+            // Re-activate transaction creation
+            TxControl.enable();
+        }
+    }
+
+    @Test
+    @BMScript("recoverySuspend")
+    public void testSuspensionWheneThereArentTxnsToRecover() {
+
+        // Make sure that the test environment is ready
+        BytemanControlledRecord.resetCommitCallCounter();
+        BytemanControlledRecord.setGreenFlag();
```

Comment:
The difference between this test and the next test is subtle, I think it's worth a comment to explain this difference here.

Reply:
Fair. Thanks Tom